### PR TITLE
Add 3.12 examples from the typing preview

### PR DESCRIPTION
**Pull request by gahjelle**
**Where to put new files:**

- New files should go into a top-level subfolder, named after the article slug. For example: `my-awesome-article`

**How to merge your changes:** 

1. [Make sure the CI code style tests all pass (+ run the automatic code formatter if necessary).](https://github.com/realpython/materials/blob/master/README.md)
2. Find an RP Team member on Slack and ask them to review & approve your PR.
3. Once the PR has one positive ("approved") review, GitHub lets you merge the PR.
4. 🎉

### CODE REVIEW

**Comment by gahjelle** [comment]
I see that the linters are crashing on the new type variable syntax. I'll investigate whether we can update linters, or if we should do what Leodanis did for the f-strings, and provide the new syntax within comments.

File: python-312/README.md [hunk 1]
```diff
@@ -136,6 +138,37 @@ Pythonista!
...
+#### Type Variables and Generic Classes, Functions, and Type Aliases
+
+You can find comparisons between the old and the new syntax for type variables in the following files:
+
+- [`generic_queue.py`](typing/generic_queue.py) and [`generic_queue_311.py`](typing/generic_queue_311.py)
+- [`list_helpers.py`](typing/list_helpers.py) and [`list_helpers_311.py`](typing/list_helpers_311.py)
+- [`concatenation.py`](typing/concatenation.py) and [`concatenation_311.py`](typing/concatenation_311.py)
+- [`inspect_string.py`](typing/inspect_string.py) and [`inspect_string_311.py`](typing/inspect_string_311.py)
+- [`deck.py`](typing/deck.py) and [`deck_311.py`](typing/deck_311.py)
+- [`alias.py`](typing/alias.py) and [`alias_311.py`](typing/alias_311.py)
```
Comment (bzaczynski) [inline python-312/README.md:160]: :warning: **Fact Check**

Missing files? I'm not seeing the `*_311.py` files in this pull request.

**Review by bzaczynski — CHANGES_REQUESTED** [review]

File: python-312/README.md (hunk 1 above)
Comment (gahjelle) [reply python-312/README.md]: Ah, sorry. I had updated the README to remove these links after I combined the 3.11 and 3.12, but the README wasn't committed. Should be fixed now.

**Review by bzaczynski — APPROVED** [review]
@gahjelle LGTM!

**Pull request merged**